test/e2e/framework/framework.go

```diff
@@ -224,59 +222,42 @@ func IsResourceAvailable(errChan chan error, client *ClientSet, resource Testing
 	errChan <- err
 }
 
-// checks 3 times if the resources are unavailable
-// - is fine if fails or 1st or 2nd run, resources could be in the process of being removed
-// - is not fine if resources disappear, then reappear
-// - it seems to take a bit longer to remove resources, so this wrapper should account for that.
-func ConsoleResourcesUnavailable(client *ClientSet) error {
-	var failed error = nil
-	// give it 3 tries, then fail
-	for i := 0; i < 3; i++ {
-		// testing resources are hard-coded in this func.
-		err := LoopResources(client, IsResourceUnavailable)
-		fmt.Printf("validating console resources have been removed... %v\n", err == nil)
-		failed = err
-	}
-	return failed
-}
-
-func LoopResources(client *ClientSet, inner func(errChan chan error, client *ClientSet, resource TestingResource)) error {
+func ConsoleResourcesUnavailable(t *testing.T, client *ClientSet) error {
 	resources := getTestingResources()
 
 	errChan := make(chan error)
 	for _, resource := range resources {
-		go inner(errChan, client, resource)
+		go IsResourceUnavailable(t, errChan, client, resource)
 	}
 	checkErr := <-errChan
 
 	return checkErr
 }
 
-// IsResourceUnavailable checks if tested resource is unavailable during a 15 second period.
-// If the resource exists during that time, an error will be returned.
-func IsResourceUnavailable(errChan chan error, client *ClientSet, resource TestingResource) {
+// IsResourceUnavailable checks if the tested resource is unavailable for a sustained period of at least ~15
+// seconds.
+func IsResourceUnavailable(t *testing.T, errChan chan error, client *ClientSet, resource TestingResource) {
 	counter := 0
 	maxCount := 15
-	err := wait.Poll(1*time.Second, AsyncOperationTimeout, func() (stop bool, err error) {
-
-		obtainedResource, err := GetResource(client, resource)
+	err := wait.PollImmediate(1*time.Second, AsyncOperationTimeout, func() (stop bool, err error) {
+		_, err = GetResource(client, resource)
 		if err == nil {
-
-			yamlBytes, err := yaml.Marshal(obtainedResource)
-			if err != nil {
-				fmt.Printf("error marshalling yaml for %s %s %v", resource.kind, resource.name, err)
-			}
-
-			return true, fmt.Errorf("deleted console %s %s was recreated: %#v", resource.kind, resource.name, string(yamlBytes))
+			// Resource exists. Reset the counter and keep waiting.
+			counter = 0
+			t.Logf("console %s %s still exists", resource.kind, resource.name)
+			return false, nil
 		}
 		if !errors.IsNotFound(err) {
-			return true, err
-		}
-		counter++
-		if counter == maxCount {
+			// Unable to retrieve resource due to error. Reset the counter and keep waiting.
+			counter = 0
+			t.Errorf("error checking that console %s %s still exists: %v", resource.kind, resource.name, err)
 			return true, nil
```

Comment:
I think that this will immediately yield - `PollImmediate tries a condition func until it returns true, an error, or the timeout is reached`

Reply:
Done